### ISSUE TITLE
Support loading Transformer Engine FP8 checkpoints and compiling with the TE executor?

### ISSUE THREAD

**Issue by ultranationalism** [Lightning-AI/lightning-thunder#2622]
## 🚀 Feature

Teach thunder.jit to parse FP8 scaling metadata from Transformer Engine & TensorRT-ModelOpt checkpoints.

### Motivation

I have FP8-trained checkpoints produced by
Transformer Engine native path:

> Keys: layer.weight (E4M3), layer._extra_state

TensorRT-ModelOpt path:

> Keys: layer.weight (E4M3), layer._amax

Thunder today only understands plain PyTorch tensors; it will happily load the files but ignores _extra_state / _amax, so the weights are treated as raw bf16/fp32 and the TE executor never receives the pre-computed scaling factors.
As a result I must re-run calibration or stay in pure-TE inference, losing Thunder’s graph optimizations.

### Pitch

Extend Thunder’s TE executor so that  

```python
model.load_state_dict(torch.load('ckpt.pt'), strict=False)
thunder.jit(model, executors=["transformer_engine_ex"])
```

recognises and **imports**  
- `_extra_state`  
- `_amax`     
 
and re-uses these values instead of re-collecting them during the delayed-scaling warm-up window.  
This would allow **zero-copy** loading of FP8 checkpoints while still utilising Thunder’s fusion pipeline.

**Comment by t-vi** [comment]
Thank you for your interest and your issue.
```
model.load_state_dict(torch.load('ckpt.pt'), strict=False)
thunder.jit(model, executors=["transformer_engine_ex"])
```
in this setup, the loading is on the module.

We do support transforming state dicts, and one could write a transform. We do not aim, however, to implement automatic recognition of third-party file formats, you'd need a conversion step.
